lib/driver.js

```diff
@@ -109,6 +109,14 @@ class IosDriver extends BaseDriver {
   async createSession (caps) {
     let [sessionId] = await super.createSession(caps);
 
+    // appium-ios-driver uses Instruments to automate the device
+    // but Xcode 8 does not have Instruments, so short circuit
+    this.xcodeVersion = await utils.getAndCheckXcodeVersion(this.opts);
+    logger.debug(`Xcode version set to ${this.xcodeVersion.versionString}`);
+    if (this.xcodeVersion.major === 8) {
+      throw new errors.SessionNotCreatedError(`appium-ios-driver does not support xcode version ${this.xcodeVersion.versionString}.`);
```

Comment:
maybe a note as to _why_ it's not supported? (`Apple has deprecated UIAutomation. Use the "XCUITest" automationName capability instead`)